doc/en/writing_plugins.rst

```diff
@@ -621,12 +621,39 @@ the new plugin:
 
 Hooks are usually declared as do-nothing functions that contain only
 documentation describing when the hook will be called and what return values
-are expected.
+are expected. The names of the functions must start with `pytest_` otherwise pytest won't recognize them.
 
 For an example, see `newhooks.py`_ from `xdist <https://github.com/pytest-dev/pytest-xdist>`_.
 
 .. _`newhooks.py`: https://github.com/pytest-dev/pytest-xdist/blob/974bd566c599dc6a9ea291838c6f226197208b46/xdist/newhooks.py
 
+To register the hooks with pytest they need to be structured in their own module or class. This
+class or module can then be passed to the `pluginmanager` using the `pytest_addhooks` function
+(which itself is a hook exposed by pytest):
+
+.. code-block:: python
+
+    def pytest_addhooks(pluginmanager):
+        """ This example assumes the hooks are grouped in the pytest_hooks module. """
+        from my_app.tests import pytest_hooks
```

Comment:
It is not a good idea to call a file `pytest_hooks`, because it will appear in the namespace of `my_app.tests` and pytest will complain that it doesn't recognize the hook `pytest_hooks` (because it starts with `pytest_`, pytest assumes it must be a hook).

I suggest using simply `hooks`.

Reply:
It is also probably show the contents of the `hooks` file, and make `pytest_my_hook` receive one or more parameters to showcase argument passing in `my_fixture` down below.